### PR TITLE
Add Ruby 3.1 to the CI matrix

### PR DESCRIPTION
**Pull request by petergoldstein**
Adds Ruby 3.1.1 to the CI matrix, and in accord with the comment removes 2.5.9 (which is EOLed)

In addition to the required change to the CI configuration, the following changes were required to get everything green:

1. Updating the `gemfile/*.lock` files to latest allowed gem and bundler versions
2. Adding `spec.metadata['rubygems_mfa_required'] = 'true'` to the gemspec to satisfy an updated Rubocop
3. Using the constants `Rack::CONTENT_TYPE` and `Rack::CACHE_CONTROL` in the `HEADERS` in `sidekiq/prometheus/exporter.rb` because `rack` now only accepts lower case versions of these strings.

This all runs green on my fork.

### CODE REVIEW

File: sidekiq-prometheus-exporter.gemspec [hunk 1]
```diff
@@ -42,4 +42,5 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rubocop-rake',              '~> 0.6'
   spec.add_development_dependency 'rubocop-rspec',             '~> 2.6'
   spec.add_development_dependency 'timecop',                   '~> 0.9'
+  spec.metadata['rubygems_mfa_required'] = 'true'
```
Comment (Strech) [inline sidekiq-prometheus-exporter.gemspec:45]: Can you please move it under the line 17, right after the `allowed_push_host`.

File: lib/sidekiq/prometheus/exporter.rb [hunk 2]
```diff
@@ -12,7 +12,7 @@ module Exporter
       NOT_FOUND_TEXT = 'Not Found'.freeze
       MOUNT_PATH = '/metrics'.freeze
       # rubocop:disable Style/MutableConstant
-      HEADERS = {'Content-Type' => 'text/plain; version=0.0.4', 'Cache-Control' => 'no-cache'}
+      HEADERS = {Rack::CONTENT_TYPE => 'text/plain; version=0.0.4', Rack::CACHE_CONTROL => 'no-cache'}
```
Comment (Strech) [inline lib/sidekiq/prometheus/exporter.rb:15]: Just to protect ourselves from non-existing constants, let's add Rack dependency like this

```ruby
spec.add_dependency 'rack', '>= 1.6.0'
```

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -30,7 +30,7 @@ jobs:
     strategy:
       matrix:
         # Always keep a window of "5 most recent" including (or on top with) HEAD
-        ruby: ["2.5.9", "2.6.8", "2.7.4", "3.0.2", "latest"]
+        ruby: ["2.6.8", "2.7.4", "3.0.2", "3.1.1", "latest"]
```
Comment (Strech) [inline .github/workflows/ci.yaml:33]: Can you please allow failures (see `exclude` section) for the new ruby version and remove `2.5.9` from that list.

**Review by Strech — CHANGES_REQUESTED** [review]
Hey @petergoldstein, thanks for your contribution, I appreciate it! I would like to ask you about few minor tweaks and we good to merge it!

**Comment by petergoldstein** [comment]
@Strech Those changes are in.

**Review by Strech — APPROVED** [review]
Amazing! Thanks a lot, I will prepare release notes 💜

**Comment by Strech** [comment]
New version of all the assets was released as [v0.1.17](https://github.com/Strech/sidekiq-prometheus-exporter/releases/tag/v0.1.17), thanks for your support 🎉

**Pull request merged**